snow/consensus/snowball/flat.go

```diff
@@ -8,16 +8,14 @@ import (
 	"github.com/ava-labs/avalanchego/utils/bag"
 )
 
-var (
-	_ Factory   = (*FlatFactory)(nil)
-	_ Consensus = (*Flat)(nil)
-)
-
-// FlatFactory implements Factory by returning a flat struct
-type FlatFactory struct{}
+var _ Consensus = (*Flat)(nil)
 
-func (FlatFactory) New() Consensus {
-	return &Flat{}
+func NewFlat(params Parameters, choice ids.ID) Consensus {
+	f := &Flat{
+		params: params,
+	}
+	f.nnarySnowball.Initialize(params.BetaVirtuous, params.BetaRogue, choice)
```

Comment:
q: why not dropping this Initialize as well?

Reply:
this is being done in a followup